wi-fi/hal/cyhal_sdio.c

```diff
@@ -296,6 +300,18 @@ static void sdio_irq_thread(void *arg)
 	cy_rtos_exit_thread();
 }
 
+/* NOTE: obj is ignored - state is kept in sdio_common */
+cy_rslt_t cyhal_sdio_start_irq_thread(cyhal_sdio_t *obj)
+{
+	cy_log_msg(CYLF_SDIO, CY_LOG_DEBUG, "cyhal_sdio_start_irq_thread\n");
+	if (cy_rtos_create_thread(&sdio_common.irq_thread_id, sdio_irq_thread, "SDIO_IRQ", NULL, 1024, 4, NULL) != CY_RSLT_SUCCESS) {
+		cy_log_msg(CYLF_SDIO, CY_LOG_ERR, "failed to start SDIO IRQ handler thread\n");
+		cyhal_sdio_free(obj);
```

Comment:
Possible double free on `sdio_obj` - check `cybsp_free` call at line 248 in `main.c`. I think you can remove `obj` from this function.